src/adapters/github/storage-layer.ts

```diff
@@ -0,0 +1,342 @@
+                owner: this.owner,
+                repo: this.repo,
+                path,
+                message: `Update ${type} data`,
+                content: Buffer.from(content).toString("base64"),
+                sha
+            });
+        } catch (er) {
+            this.logger.error("Failed to store data", { er });
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Does not fetch granular data, instead obtains the DB object
+     * and returns it.
+     * 
+     * Fitted with a helper for returning the correct storage type depending on the param.
+     */
+    async retrieveStorageDataObject<TType extends StorageTypes = StorageTypes>(type: TType, withSha?: boolean): Promise<RetrievalHelper<TType>> {
+        let path;
+        switch (type) {
+            case "singleChat":
+            case "allChats":
+                path = this.chatStoragePath;
+                break;
+            case "userBank":
+                path = this.userBankPath;
+                break;
+            case "session":
+                path = this.telegramSessionPath;
+                break;
+            default:
+                throw new Error("Invalid storage type");
+        }
+
+        const { data } = await this.octokit.repos.getContent({
+            owner: this.owner,
+            repo: this.repo,
+            path
+        });
+
+        let dataContent;
+
+        if ("content" in data) {
+            dataContent = data.content;
+        } else {
+            throw new Error("Failed to retrieve data content");
+        }
+
+        const content = Buffer.from(dataContent, "base64").toString();
+
+        let parsedData: RetrievalHelper<TType>;
+
+        try {
+            parsedData = JSON.parse(content);
+        } catch (e) {
+            throw new Error("Failed to parse JSON data");
+        }
+
+        return {
+            ...parsedData,
+            ...(withSha ? { sha: data.sha } : {})
+        }
+    }
+}
+
+function deleteAllShas<T extends Withsha>(data: T) {
+    Object.keys(data).forEach((key) => {
+        const value = data[key as keyof typeof data];
+
+        if (key === "sha") {
+            Reflect.deleteProperty(data, key);
+        }
+
+        if (typeof value === "object" && value) {
+            deleteAllShas(value);
+        }
+
+        if (Array.isArray(value)) {
+            value.forEach((item) => {
+                if (typeof item === "object" && item) {
+                    deleteAllShas(item);
+                }
+            })
+        }
+    });
+
+    return data
+}
+
+function isSessionStorage(data: unknown): data is SessionStorage {
+    if (typeof data !== "object" || !data) return false;
+    return "session" in data;
+}
+
+function isChatsStorage(data: unknown): data is ChatStorage {
+    if (typeof data !== "object" || !data) return false;
+    return "chats" in data;
+}
+
+function isUserBankStorage(data: unknown): data is UserBank {
```

Comment:
What is bank

Reply:
I can rename to whatever is more appropriate but it's this image below:

What else do you think I should record? Wallet address? anything else?

![image](https://github.com/user-attachments/assets/620c3b39-809f-45bc-a9d7-e5e744ef7d46)

Reply:
added `walletAddress` and a basic `/wallet` command to add/update it